File: mvp-framework/src/main/java/com/stephen/mvpframework/annotation/InjectLayoutId.java
package com.stephen.mvpframework.annotation;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 布局ID注入注解
 * Created by Stephen on 2017/10/4.
 * StephenYoung0406@hotmail.com
 * <(￣ c￣)y▂ξ
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface InjectLayoutId {
    //布局ID
    int ID() default 0;
}

File: mvp-framework/src/main/java/com/stephen/mvpframework/annotation/InjectLayoutId.kt
package com.stephen.mvpframework.annotation

/**
 * 布局ID注入注解
 * Created by Stephen on 2018/11/15.
 * StephenYoung0406@hotmail.com
 * <(￣ c￣)y▂ξ
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class InjectLayoutId(val ID: Int = 0)